service/src/main/java/bio/terra/profile/service/azure/AzureService.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.profile.service.azure;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.profile.model.AzureManagedAppModel;
+import bio.terra.profile.service.crl.CrlService;
+import com.azure.resourcemanager.managedapplications.models.Application;
+import java.util.List;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AzureService {
+  private static final Logger logger = LoggerFactory.getLogger(AzureService.class);
+  private static final String AUTHORIZED_USER_KEY = "authorizedTerraUser";
+  // TODO externalize this
+  private static final String TERRA_APP_PRDDUCT = "fake-terra-app-product-name";
+
+  private final CrlService crlService;
+
+  @Autowired
+  public AzureService(CrlService crlService) {
+    this.crlService = crlService;
+  }
+
+  /**
+   * Gets the Azure managed applications the user has access to in the given subscription.
+   *
+   * @param subscriptionId Azure subscription ID that will be checked for managed applications
+   * @param userRequest Authorized user request
+   * @return List of Terra Azure managed applications the user has access to
+   */
+  public List<AzureManagedAppModel> getManagedAppDeployments(
+      UUID subscriptionId, AuthenticatedUserRequest userRequest) {
+    var appMgr = this.crlService.getApplicationManager(subscriptionId);
+    return appMgr.applications().list().stream()
+        .filter(app -> isAuthedTerraManagedApp(userRequest, app))
+        .map(app -> new AzureManagedAppModel().name(app.name()))
+        .toList();
```

Comment:
I read in JDK17 you didn't need `.collect(Collectors.toList())` to convert a Stream to a List but it didn't work in my IDE ;-)

Reply:
The form here (`.toList()`) is the new one. If it doesn't work in the IDE maybe you have a project setting that isn't enabling JDK 17 features?